sdcm/utils/compaction_ops.py

```diff
@@ -0,0 +1,103 @@
+import time
+from typing import Callable, Optional, NamedTuple, Union
+
+from fabric.runners import Result
+
+from sct import LOGGER
+from sdcm.cluster import BaseNode, BaseCluster, BaseScyllaCluster
+from sdcm.rest.storage_service_client import StorageServiceClient
+
+
+class ScrubModes(NamedTuple):
+    ABORT: str = "ABORT"
+    SKIP: str = "SKIP"
+    SEGREGATE: str = "SEGREGATE"
+    VALIDATE: str = "VALIDATE"
+
+
+class NodetoolCommands(NamedTuple):
+    flush: str = "flush"
+    stop_major_compaction: str = "stop COMPACTION"
+    stop_scrub_compaction: str = "stop SCRUB"
+    stop_cleanup_compaction: str = "stop CLEANUP"
+    stop_upgrade_compaction: str = "stop UPGRADE"
+    stop_reshape_compaction: str = "stop RESHAPE"
+
+
+class CompactionOps:
+    NODETOOL_CMD = NodetoolCommands()
+    SCRUB_MODES = ScrubModes()
+
+    def __init__(self, cluster: Union[BaseCluster, BaseScyllaCluster], node: Optional[BaseNode] = None):
+        self.cluster = cluster
+        self.node = node if node else self.cluster.nodes[0]
+        self.storage_service_client = StorageServiceClient(node=self.node)
+
+    def trigger_major_compaction(self, keyspace: str = "keyspace1", cf: str = "standard1") -> Result:
```

Comment:
@KnifeyMoloko as I commented below, we need it to run on any keyspace & table, not only keyspace1.standard1.
Since it's a very big PR, I don't mind to merge it and you will send a following PR to update this.
What you can do is to query all non system tables (we have such a method) and pick one randomly each time.